pkg/virt-controller/watch/vmi/vmi.go

```diff
@@ -2175,9 +2175,8 @@ func (c *Controller) getVolumePhaseMessageReason(volume *virtv1.Volume, namespac
 func (c *Controller) updateMultusAnnotation(namespace string, interfaces []virtv1.Interface, networks []virtv1.Network, pod *k8sv1.Pod) error {
 	podAnnotations := pod.GetAnnotations()
 
-	indexedMultusStatusIfaces := multus.NonDefaultNetworkStatusIndexedByPodIfaceName(multus.NetworkStatusesFromPod(pod))
-	networkToPodIfaceMap := namescheme.CreateNetworkNameSchemeByPodNetworkStatus(networks, indexedMultusStatusIfaces)
-	multusAnnotations, err := multus.GenerateCNIAnnotationFromNameScheme(namespace, interfaces, networks, networkToPodIfaceMap, c.clusterConfig)
+	podIfaceNamesByNetworkName := namescheme.CreateFromNetworkStatuses(networks, multus.NetworkStatusesFromPod(pod))
+	multusAnnotations, err := multus.GenerateCNIAnnotationFromNameScheme(namespace, interfaces, networks, podIfaceNamesByNetworkName, c.clusterConfig)
```

Comment:
Is there a scenario where it is possible for its `networkNameScheme` parameter to be empty?

Currently, it is called from:
1. `GenerateCNIAnnotation` with hashed naming scheme.
2. `Generator.GenerateFromSource` with ordinal naming scheme.
3. `Controller.updateMultusAnnotation` with hashed or ordinal naming scheme.

BTW, it is not covered by unit tests.

Reply:
I think it can happen in case Multus network-status annotation is maflormed/invalid, right?
Before the dynamic-nic-name PR changed, `GenerateCNIAnnotationFromNameScheme` would recive name-scheme map that is non-empty.
I suppose this is why we had no test for that, there was no scenario where the mapping is empty.

Reply:
Anyhow, it can be done on a follow-up PR because it didnt caused by this PR changed, sorry for the noise.